### DISCUSSION TITLE
want to enable ampqs :5671 in ingress

### DISCUSSION THREAD

**Discussion by jags06** [rabbitmq/rabbitmq-server#2809] [Questions]
Hi 
i am using helm char to deploy rabbitmq in cluster 
https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/values.yaml
using 
****
##
  tls:
    enabled: true
    failIfNoPeerCert: false
    sslOptionsVerify: verify_none
existingSecret:xxxxx
existingSecretFullChain: true

i am already having NLB and using ingress i can communicate to 5672 for communication but want to enable tls communication over 5671 with port ampqs and want to take leverage of certificate rotation by Lets-encrypt in cluster. Can you guide me how can i enable it 

i have tried this echo "apiVersion: configuration.konghq.com/v1beta1
kind: TCPIngress
metadata:
  name: rabbit-release-rabbitmq
  annotations:
    kubernetes.io/ingress.class: kong
spec:
  entryPoints:
    - amqps
  routes:
    - match: HostSNI(`xxxxxxx.com`)  
  rules:
  - port: 5671
    backend:
      serviceName: rabbit-release-rabbitmq
      servicePort: 5671 
  tls:
  - secretName: xxxxxxx.com-tls
    hosts:
    - rabbitmq.plnlondon.com
    passthrough: true     
" | kubectl apply -f -

if i change the servicePort: 5672
then i can receive data over ampqs from client as well . is this correct way to handle traffic over tls ?

**Comment by ChunyiLyu** [comment]
The RabbitMQ team does not maintain [this helm chart](https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq).

We have a [RabbitMQ cluster operator](https://github.com/rabbitmq/cluster-operator) that has TLS support. Here are some examples/docs that you can check out:

1. [different tls Configurations](https://www.rabbitmq.com/kubernetes/operator/using-operator.html#tls-conf)
2. [tls example](https://github.com/rabbitmq/cluster-operator/tree/main/docs/examples/tls)
3. [mtls example](https://github.com/rabbitmq/cluster-operator/tree/main/docs/examples/mtls)
4. [inter-node tls example](https://github.com/rabbitmq/cluster-operator/tree/main/docs/examples/mtls-inter-node)
